flink-libraries/flink-table/src/main/scala/org/apache/flink/table/plan/rules/dataSet/DataSetAggregateRule.scala

```diff
@@ -42,6 +41,14 @@ class DataSetAggregateRule
       return false
     }
 
+    // distinct is translated into dedicated operator
+    if (agg.getAggCallList.isEmpty &&
+      agg.getGroupCount == agg.getRowType.getFieldCount &&
```

Comment:
During testing I observed a case where a projection wasn't pushed down. 
Without this check, the grouping would happen on a subset of fields and only those would be emitted. It is not possible to change the input and output types of a ReduceFunction (which is used to implement Distinct in a hash-combinable way), I check that the input and output types are identical.